### PR TITLE
Jack/update documentation

### PR DESCRIPTION
**Pull request by jackretterer**
Updated:

- Added back support document types for partitioning
- Added more tabs for python code in the API page
- Added a RAG section in Key Concepts
- Added a Common Use case section in overview

### CODE REVIEW

**Comment by cragwolfe** [comment]
i think we're due for a release, especially with the element ordering change with fast + hi_res strategies for images and PDF's.
may as well make this a release? (remove -devX from the top line of the changelog, edit unstructured/__version__.py to be consistent)

File: docs/source/introduction/overview.rst [hunk 1]
```diff
@@ -18,14 +27,13 @@ Key Features
...
-- **Connectors**: Interfaces that enable the library to interact with different data sources and sinks, like cloud storage or databases.
-
-- **Bricks**: Modular units of the library that allow users to partition, clean, and stage data efficiently.
+Common Use Cases
+^^^^^^^^^^^^^^^^
 
-- **Metadata**: Data about data. In ``unstructured``, metadata helps in keeping track of the source, type, and other essential attributes of the data.
+- **Pretraining Models**
+- **Fine-tuning Models**
+- **Retrieval Augmented Generation (RAG)**
```
Comment (cragwolfe) [inline docs/source/introduction/overview.rst:35]: perhaps this should be top of the list?

**Comment by cragwolfe** [comment]
Looks like a markdown format issue for one row:

image

File: docs/source/bricks/partition.rst [hunk 2]
```diff
@@ -13,6 +13,55 @@ The easiest way to partition documents in unstructured is to use the ``partition
...
++----------------------------------------------+---------------------+----------------------------------------+----------------+------------------------------------------------------------------------------------------------------------------+
+| Open Office Documents (`.odt`)               | `partition_odt`     | N/A                                    | Yes            | None                                                                                                             |
++----------------------------------------------+---------------------+----------------------------------------+----------------+------------------------------------------------------------------------------------------------------------------+
+| PDFs (`.pdf`)                                | `partition_pdf`     | `"auto"`, `"fast"`, `"hi_res"`, `"ocr_only"` | Yes | Encoding; Include Page Breaks; Infer Table Structure; Max Partition; OCR Languages, Strategy                          |
```
Comment (ron-unstructured) [inline docs/source/bricks/partition.rst:43]: the table alignment is off

**Review by ron-unstructured — CHANGES_REQUESTED** [review]
Hi @jackretterer, please resolve the comments before we merge. Thanks.

**Review by ron-unstructured — APPROVED** [review]

**Pull request merged**